CRAN 0.2.2

DominiqueMakowski opened the pull request:
@strengejacke @mattansb @IndrajeetPatil @humanfactors @pdwaggoner 

Dear all we are at the brink of updating the CRAN version of bayestestR. Do not hesitate to go through the documentation/vignettes (in particular [this](https://easystats.github.io/bayestestR/articles/example1.html), [this](https://easystats.github.io/bayestestR/articles/example2.html) and [this](https://easystats.github.io/bayestestR/articles/guidelines.html) vignette) and suggest/make any changes/rephrasings/corrections etc.

Thanks a lot ☺️ 

### Issues

- [x] Formatting of rules of thumb in guidelines vignette
- [x] Fix https://github.com/easystats/bayestestR/issues/161
- [ ] ~Fix https://github.com/easystats/bayestestR/issues/168~ 🥇  :-D 😬

strengejacke commented:
Any ideas why the check fail on Travis-Linux?

strengejacke commented:
1. [Exampe 2](https://easystats.github.io/bayestestR/articles/example2.html#bayesian-correlation): _bayestestR::bayesfactor(result)_ 

The output is missing, is there any reason for this? If this is intended, we might remove the line with `bayestestR::bayesfactor(result)`.

strengejacke commented:
[in-Depth-Comparison](https://easystats.github.io/bayestestR/articles/indicesExistenceComparison.html#relationship-with-the-frequentist-p-value)

The dots have a very low alpha-value, I can hardly see any dots.

strengejacke commented:
1. [Guidelines](https://easystats.github.io/bayestestR/articles/guidelines.html#uncertainty): _We also recommend computing the CI based on the HDI_

I would add an exception, in case point estimates are transformed (e.g. exponentiated, to get something like _odds ratios_). Here, if you exponentiate the CIs as well, you may get intervals that cover the null, although they did not cover the null if left on linear scale (Jensen's inequality)

2. [Guidelines](https://easystats.github.io/bayestestR/articles/guidelines.html#existence)

Strange formatted code-chunk. I guess it should be normal text, not code.

3. [Guidelines](https://easystats.github.io/bayestestR/articles/guidelines.html#significance)

See 2.

DominiqueMakowski commented:
> Any ideas why the checks fail on Travis-Linux?

Not *really*, but I would say "because of the C code in rstan" (my default answer)

> Example 1: This distribution represents the probability (the y axis)...

The goal is to convey the "meaning". Although these values refer to the density function, they *represent* the probability... Nevertheless, I wouldn't want to make it too complicated as the important notion here is just that it is *sort of* the probability... 

I made it clearer by adding a link to the [probability density function](https://en.wikipedia.org/wiki/Probability_density_function.

And this has *nothing* to do with #168 😠 

> Example 1: Well, this is close to the mean.

fixed

> 3a. Example 1: It will give us the range containing the 89% more probable effect values.

fixed

> Example 1: (because reasons)

fixed
> 3b: (McElreath 2018)

On scholar, the book title points to (a reedition? a preprint?) of  2018...

> Exampe 2: bayestestR::bayesfactor(result)

good catch.

> The dots have a very low alpha-value, I can hardly see any dots.

Yes this is expected. This is currently the case because of the preliminary nature of the data used. However, once I have a greenlight for the generating code (in circus), I will run the simulation for 36k datapoints. Then, the alpha value is appropriate. Didn't want to change back and forth and adjust for preliminary data, as it took me some time to find an appropriate alpha value

> Guidelines: We also recommend computing the CI based on the HDI
>  I would add an exception, in case point estimates are transformed (e.g. exponentiated, to get something like odds ratios). Here, if you exponentiate the CIs as well, you may get intervals that cover the null, although they did not cover the null if left on linear scale (Jensen's inequality)

fixed. Maybe we should write a blogpost about this.

> Guidelines: Strange formatted code-chunk. I guess it should be normal text, not code.

fixing.

humanfactors commented:
Apologies here for detracting with a workflow question - but I'm struggling to work out the best way to make suggestions to the pull request for writing. I'm actually happy to work through the entire document, and make suggestions as I would any other manuscript. I'm not sure whether the best approach is to add commits in a separate branch you can pull from and merge, or to manually paste in what I'd suggest to change here in the PR directly?

DominiqueMakowski commented:
@humanfactors It's mostly up to you as it depends on your setup.

The best way IMO, if you have a desktop github installed on your machine, is to select the dev branch, make your changes there, and push (they will arrive directly on these PRs). We can then comment, review, change them etc. before merging to master. 

You can also make changes on your side and open a new PR. 

Another option, if you're not on a computer or don't have git installed on your machine, is just to make comments here or open issues (see for instance Daniel's comments above).

There is no canonical way of doing things in easystats yet, we are all mostly experiencing, trying things out and see how that works ☺️

strengejacke commented:
All native speaker can push directly into master ;-)

DominiqueMakowski commented:
> All native speaker can push directly into master ;-)

@strengejacke I did not know the repository was in German then 😁

strengejacke commented:
I would keep Travis Linux activated because in case it would work again, it helps us detecting check issues...

DominiqueMakowski commented:
> I would keep Travis Linux activated because in case it would work again, it helps us detecting check issues...

I will reactivate it as soon as it is merged (the goal being to having a passing setup on master and here we can keep it activated, and deactivating it just before merging)

strengejacke commented:
> having a passing setup on master

You and your obsession to a passing Travis on master branch :-D
This won't help us with CRAN submissions!

DominiqueMakowski commented:
> This won't help us with CRAN submissions!

True, but it is reassuring... And don't you see I am trying to apply CBT to cure myself, but as we know autotherapy does not work well, so be patient *and make master pass travis* 😅 

Besides, who uses linux anyway... 😬

strengejacke commented:
> Besides, who uses linux anyway... 😬

True, but same holds true for OS X, so let's remove Travis entirely, so you're no longer a slave of an unreliable "therapist" :-)

DominiqueMakowski commented:
Alright, I'll remove travis as soon as you start pushing to dev branches 😄😉

DominiqueMakowski commented:
Let's hold on for tonight so that the west side people can have a look if they want. Anyway, the plan is to merge, - and submit, tomorrow ☺️

strengejacke commented:
I think Linux on Travis is back again, so you may run the Travis-tests for Linux as well... There have been temporary problems with rstanarm on CRAN.

DominiqueMakowski commented:
Okay let's try give me one sec

strengejacke commented:
It would be great if Travis Linux would work _and_ the package passes the check, because that was one of the crucial parts on CRAN check side.

DominiqueMakowski commented:
> It would be great if Travis Linux would work and the package passes the check, because that was one of the crucial parts on CRAN check side.

Couldn't agree more 😅

DominiqueMakowski commented:
Submitted to CRAN 🤞

strengejacke commented:
Btw, see 0.2.0 is on CRAN!

strengejacke commented:
Haven't you merged this PR for the CRAN submission?

DominiqueMakowski commented:
no the idea was to merge it right after CRAN acceptance

strengejacke commented:
Ok. I thought it would make sense to have the revisions and corrections in the CRAN release?

DominiqueMakowski commented:
Yes it is in the CRAN release, I am submitting this branch to CRAN

strengejacke commented on vignettes/indicesExistenceComparison.Rmd:
```diff
@@ -206,7 +206,7 @@ df %>%
   mutate(temp = as.factor(cut(sample_size, 3, labels = FALSE))) %>% 
   group_by(temp) %>% 
-  mutate(size_group = as.factor(round(mean(sample_size)))) %>% 
+  mutate(size_group = as.character(round(mean(sample_size)))) %>% 
```
I'm, not sure if you need a factor for proper sorting in the plot, but as not all values of `sample_size` are present in all groups, `mutate()` coerces to character, anyway, so this avoids the warning.

If a factor is required, you need to specify _all_ available values from `sample_size` in the `levels`-argument, I guess (and hence, you need something like `all_levels <- unique(df$sample_size)` before the pipe-chain).

DominiqueMakowski commented:
please feel free to make any improvements that you find relevant!

DominiqueMakowski commented:
Submitted (2nd try)

DominiqueMakowski commented:
That was quick! Great job everyone ☺️ 
```
Dear maintainer,

thanks, package bayestestR_0.2.2.tar.gz is on its way to CRAN.

Best regards,
CRAN teams' auto-check service
Package check result: OK

No changes to worse in reverse depends.
```

DominiqueMakowski commented on .travis.yml:
```diff
@@ -2,13 +2,13 @@ language: r
   #- windows
-  - linux
-  # - osx
+  #- linux
```
I meant "deactivate linux"
